contracts/jobs/detached/V2DetachedJob.sol

```diff
@@ -0,0 +1,211 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity >=0.8.4 <0.9.0;
+
+import '@openzeppelin/contracts/utils/structs/EnumerableSet.sol';
+import '@lbertenasco/contract-utils/contracts/abstract/MachineryReady.sol';
+
+import '../../interfaces/jobs/v2/IV2Keeper.sol';
+import '../../interfaces/jobs/detached/IV2DetachedJob.sol';
+
+import '../../interfaces/yearn/IBaseStrategy.sol';
+import '../../interfaces/oracle/IYOracle.sol';
+import '../../interfaces/utils/IBaseFee.sol';
+
+abstract contract V2DetachedJob is MachineryReady, IV2DetachedJob {
+  using EnumerableSet for EnumerableSet.AddressSet;
+
+  address public constant WETH = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+  address public immutable baseFeeOracle;
```

Comment:
Why are we using this externally ? It adds to the gas costs probably. Doesn't matter in FTM tho ... But mainnet, probably does.